src/main/java/ch/njol/skript/effects/EffEnchant.java

```diff
@@ -53,55 +53,51 @@ public class EffEnchant extends Effect {
 	}
 	
 	@SuppressWarnings("null")
-	private Expression<ItemType> item;
+	private Expression<ItemType> items;
 	@Nullable
-	private Expression<EnchantmentType> enchs;
+	private Expression<EnchantmentType> enchantments;
 	
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		item = (Expression<ItemType>) exprs[0];
-		if (!ChangerUtils.acceptsChange(item, ChangeMode.SET, ItemStack.class)) {
-			Skript.error(item + " cannot be changed, thus it cannot be (dis)enchanted");
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		items = (Expression<ItemType>) exprs[0];
+		if (!ChangerUtils.acceptsChange(items, ChangeMode.SET, ItemStack.class)) {
+			Skript.error(items + " cannot be changed, thus it cannot be (dis)enchanted");
 			return false;
 		}
 		if (matchedPattern == 0)
-			enchs = (Expression<EnchantmentType>) exprs[1];
+			enchantments = (Expression<EnchantmentType>) exprs[1];
 		return true;
 	}
 	
 	@Override
-	protected void execute(final Event e) {
-		final ItemType i = item.getSingle(e);
-		if (i == null)
+	protected void execute(Event event) {
+		ItemType[] items = this.items.getArray(event);
+		if (items.length == 0) // short circuit
 			return;
-		if (enchs != null) {
-			final EnchantmentType[] types = enchs.getArray(e);
+
+		if (enchantments != null) {
+			EnchantmentType[] types = enchantments.getArray(event);
 			if (types.length == 0)
 				return;
-			
-			for (final EnchantmentType type : types) {
-				Enchantment ench = type.getType();
-				assert ench != null;
-				i.addEnchantments(new EnchantmentType(ench, type.getLevel()));
+			for (ItemType item : items) {
+				for (EnchantmentType type : types) {
+					Enchantment enchantment = type.getType();
+					assert enchantment != null;
```

Comment:
An assertion means nothing here without a testing environment. Potentially add tests to this pull request.